pkg/expr/ml.go

```diff
@@ -61,21 +62,6 @@ func (m *MLNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s *
 	var result mathexp.Results
 	timeRange := m.TimeRange.AbsoluteTime(now)
 
-	// get the plugin configuration that will be used by client (auth, host, etc)
-	pCtx, err := s.pCtxProvider.Get(ctx, mlPluginID, m.request.User, m.request.OrgId)
-	if err != nil {
-		if errors.Is(err, plugins.ErrPluginNotRegistered) {
-			return result, errMLPluginDoesNotExist
-		}
-		return result, fmt.Errorf("failed to get plugin settings: %w", err)
-	}
-
-	// Plugin must be initialized by the admin first. That will create service account, and update plugin settings so all requests can use it.
-	// Fail if it is not initialized.
-	if pCtx.AppInstanceSettings == nil || !jsoniter.Get(pCtx.AppInstanceSettings.JSONData, "initialized").ToBool() {
```

Comment:
@yuri-tceretian pinging you since you introduced this in #69963. This logic feels a bit misplaced here/unoptimal when you're calling the ml-app plugin. Did you consider letting the ml app handle this logic?

Reply:
ML plugin sets the initialized flag when a user (admin?) logs in, goes to the ML page, and clicks a button. The plugin is installed at that moment. When the user clicks the button, the ML plugin (UI portion) registers an API key and updates the plugin settings. 
Alerting, where this expression is used, is a background service. There is a chance that a rule is created and evaluated when the ML plugin happens to be not initialized. For example, provisioning that is run right after Grafana starts. So, the idea is to let the user create alert rules using ML plugin but fail until the user initializes it.